suitcase/spec.py

```diff
@@ -340,13 +384,23 @@ def __str__(self):
 {} """.format(self.scan_id, self.scan_command + " " + " ".join(self.scan_args),
               len(self), self.time_from_date)
 
+    # def __hash__(self):
+    #     return int(_md5(self.raw_scan_data)
 
+def _md5(iterable):
+    if not isinstance(iterable, six.string_types):
+        iterable = '\n'.join(iterable)
+    if isinstance(iterable, six.text_type):
+        iterable = iterable.encode()
+    return hashlib.md5(iterable).hexdigest()
 
 ###############################################################################
 # Spec to document code
 ###############################################################################
 
-def spec_to_document(specfile, scan_ids=None, validate=False):
+@singledispatch
```

Comment:
Thanks!  Found that `singledispatch` was described very clearly in the Fluent Python book and have been looking for places to use it ever since